lib/internal/process/next_tick.js

```diff
@@ -56,15 +18,47 @@ function setupNextTick() {
   // Grab the constants necessary for working with internal arrays.
   const { kInit, kDestroy, kAsyncIdCounter } = async_wrap.constants;
   const { async_id_symbol, trigger_async_id_symbol } = async_wrap;
-  const nextTickQueue = new NextTickQueue();
-  var microtasksScheduled = false;
 
-  // Used to run V8's micro task queue.
-  var _runMicrotasks = {};
+  // tickInfo is used so that the C++ code in src/node.cc can
+  // have easy access to our nextTick state, and avoid unnecessary
+  // calls into JS land.
+  // runMicrotasks is used to run V8's micro task queue.
+  const [
+    tickInfo,
+    runMicrotasks
+  ] = process._setupNextTick(_tickCallback);
 
   // *Must* match Environment::TickInfo::Fields in src/env.h.
-  var kIndex = 0;
-  var kLength = 1;
+  const kScheduled = 0;
+
+  const nextTickQueue = {
```

Comment:
I'm curious, why make this an object literal instead of the class that existed before? I would think V8 still optimizes classes (using hidden classes) better than object literals.

Reply:
AFAIK, It does the same thing with both in terms of hidden classes. (That's part of the reason `Object.create(null)` is used in a lot of our code because it prevents creating a hidden class and just skips straight to dictionary object mode.)

The reasoning for the change is mostly that it's straight-up confusing the first time one looks at this code, `NextTickQueue` is declared outside of `setup` and it appears that multiple queues might be supported until one starts reading into the source.

Reply:
Also, the `tickInfo[kScheduled]` flag is flipped inside `push` and `shift` now, so declaring it outside of `setup` is impossible.

Reply:
I see. Properly we could merge the code for the `timers` linked list and the `nextTick` linked list, but this is fine for now.

Reply:
Ideally, we wouldn't. `nextTick` has a special version that's optimized for its own use because it only needs a singly linked list, whereas timers definitely need doubly linked.

Reply:
We have very different concepts of what is ideal :)

Reply:
Using doubly linked list, despite it being completely unnecessary for nextTick, would come with a significant performance regression and memory consumption increase...